Sources/Build/BuildPlan.swift

```diff
@@ -563,12 +561,8 @@ public class BuildPlan: SPMBuildCore.BuildPlan {
     ) throws {
         // Supported platforms are defined at the package level.
         // This will need to become a bit complicated once we have target-level or product-level platform support.
-        guard let productPlatform = product.platforms.getDerived(for: .macOS) else {
-            throw StringError("Expected supported platform macOS in product \(product)")
-        }
-        guard let targetPlatform = target.platforms.getDerived(for: .macOS) else {
-            throw StringError("Expected supported platform macOS in target \(target)")
-        }
+        let productPlatform = product.platforms.getDerived(for: .macOS, usingXCTest: product.type == .test)
+        let targetPlatform = target.platforms.getDerived(for: .macOS, usingXCTest: product.type == .test)
```

Comment:
guards not needed any longer?

Reply:
that's right, there will always be a value now